docs/src/main/asciidoc/http-reference.adoc

```diff
@@ -431,6 +431,13 @@ include::{generated-dir}/config/quarkus-vertx-http-config-group-access-log-confi
 Set `quarkus.http.record-request-start-time=true` to enable recording request start times when using any of the attributes related to logging request processing times.
 ====
 
+[NOTE]
+====
+Assuming security has been set up for the application (see our xref:security-overview.adoc[guide] for more details),
```

Comment:
Thanks Michal but I'm not sure it helps in the case where the user is using a pre-matching server filter: `@ServerRequestFilter(preMatching = true, priority = 2, nonBlocking = true)`. 
For JAX-RS we have this hint, https://quarkus.io/guides/security-customization#jaxrs-security-context, but not sure if a similar approach works for Reactive

Reply:
I played with the reproducer, I realize what the issue is about.

> https://quarkus.io/guides/security-customization#jaxrs-security-context

That doesn't work either, for RESTEasy Classic or Reactive. `jakarta.ws.rs.core.SecurityContext` is Jakarta REST concept that is not part of Quarkus Security. If you do https://quarkus.io/guides/security-customization#jaxrs-security-context it has no effect on injected identity, and you can only access these changes with `@Context SecurityContext sec` in JAX-RS resources.

I'm not sure what is your suggestion, Vert.x feature shouldn't rely on `jakarta.ws.rs.core.SecurityContext` if there is `user` on `RoutingContext` event.

Reply:
@sberyozkin I wonder if I didn't document it wrong considering how you understood it. What I tried to say is that you should use Quarkus Security (provide `SecurityIdentity`, augment `SecurityIdentity`) because then, `RoutingContexct` user should be setup correctly, therefore logging will be correct.

I'd also like to pointout this is HTTP guide, not JAX-RS guide.

Reply:
> The idea there is that there is a post match quarkus-resteasy filter which, if it spots that a JAX-RS security context is available, then it will align it with SecurityIdentity, Stuart worked on it. If it does not work now then it is a regression

You are right @sberyozkin, I think I understand you now.

If I read code correcly  when `io.quarkus.resteasy.runtime.SecurityContextFilter` updates `SecurityIdentity` it doesn't update `RoutingContext#user`. I'll try to reproduce such issue tomorrow. I can fix then in case I read it correctly.

Similarly in RESTEasy Reactive when io.quarkus.resteasy.reactive.server.runtime.security.SecurityContextOverrideHandler is run, I can't see it update same identity, therefore it won't have effect to logging.

I will look tomorrow if that is valid issue, but that won't fix #35265 , because all these features rely on Quarkus Security. And how I understood reproducer, there is none.

I think what is documented in this PR is correct. Let's discuss it only in relation to linked issues.

Reply:
Anyway, in case you don't find text documented in this PR helpful, I absolutely accept it, I have no problem to drop the PR. Thanks

Reply:
Let me phrase it differently, I think these security filters Stuart added are only present when Security extension is present. However I think it makes sense to fix it for cases when the extension is present and that would be valid issue. If we can treat #35265 like that (which it is not), I'll make this draft.

Reply:
In case of RESTEsy Reactive this augmentation of `SecurityIdentity` doesn't happen as there is no `io.quarkus.security.identity.CurrentIdentityAssociation`, but that just consequence of missing Quarkus Security.

I hope we understand each other now, I think you were right, there is probably actual bug, I just don't think it is the one described by the reproducer.